contrib/inventory/ec2.py

```diff
@@ -409,88 +437,40 @@ def read_settings(self):
         self.cache_path_index = os.path.join(cache_dir, "%s.index" % cache_name)
         self.cache_max_age = config.getint('ec2', 'cache_max_age')
 
-        if config.has_option('ec2', 'expand_csv_tags'):
-            self.expand_csv_tags = config.getboolean('ec2', 'expand_csv_tags')
-        else:
-            self.expand_csv_tags = False
+        self.expand_csv_tags = config.getboolean('ec2', 'expand_csv_tags')
 
         # Configure nested groups instead of flat namespace.
-        if config.has_option('ec2', 'nested_groups'):
-            self.nested_groups = config.getboolean('ec2', 'nested_groups')
-        else:
-            self.nested_groups = False
+        self.nested_groups = config.getboolean('ec2', 'nested_groups')
 
         # Replace dash or not in group names
-        if config.has_option('ec2', 'replace_dash_in_groups'):
-            self.replace_dash_in_groups = config.getboolean('ec2', 'replace_dash_in_groups')
-        else:
-            self.replace_dash_in_groups = True
+        self.replace_dash_in_groups = config.getboolean('ec2', 'replace_dash_in_groups')
 
         # IAM role to assume for connection
-        if config.has_option('ec2', 'iam_role'):
-            self.iam_role = config.get('ec2', 'iam_role')
-        else:
-            self.iam_role = None
+        self.iam_role = config.get('ec2', 'iam_role')
 
         # Configure which groups should be created.
-        group_by_options = [
-            'group_by_instance_id',
-            'group_by_region',
-            'group_by_availability_zone',
-            'group_by_ami_id',
-            'group_by_instance_type',
-            'group_by_instance_state',
-            'group_by_platform',
-            'group_by_key_pair',
-            'group_by_vpc_id',
-            'group_by_security_group',
-            'group_by_tag_keys',
-            'group_by_tag_none',
-            'group_by_route53_names',
-            'group_by_rds_engine',
-            'group_by_rds_parameter_group',
-            'group_by_elasticache_engine',
-            'group_by_elasticache_cluster',
-            'group_by_elasticache_parameter_group',
-            'group_by_elasticache_replication_group',
-            'group_by_aws_account',
-        ]
+
+        group_by_options = [a for a in DEFAULTS if a.startswith('group_by')]
         for option in group_by_options:
-            if config.has_option('ec2', option):
-                setattr(self, option, config.getboolean('ec2', option))
-            else:
-                setattr(self, option, True)
+            setattr(self, option, config.getboolean('ec2', option))
```

Comment:
Looks good to me!

This is working as expected.

A couple minor things I noticed:
* Using STS credentials doesn't work with dynamic inventory (it doesn't work on devel either).
* I'm also wondering about this bit. I'm reading the removed lines 459-462 as 'if this group_by_thing doesn't have the option provided it defaults to true' - if the file doesn't exist then it seems like the behavior is changing a little. (such as, group_by_aws_account is false by default in the ini but if it isn't provided it defaults to true here; now it defaults to false always). I think that's okay though.

Reply:
I transfered the defaults in the `ini` file to the `DEFAULTS` variable. The two `group_by` options that are `False` in the `ini` file are false in `DEFAULTS`. All other `group_by` options are `True`.

See lines [174](https://github.com/ansible/ansible/pull/28375/files#diff-c2e371537339bbab5d9d0cc5d6cb7adcR174) and [180](https://github.com/ansible/ansible/pull/28375/files#diff-c2e371537339bbab5d9d0cc5d6cb7adcR180).

Reply:
Yeah, I see that you kept the behavior specified by default in the ini. I just meant that before, if the group_by option was not provided those by-default Falses would default-default to True by omission (as opposed to now, when it is always False by omission) on the deleted line 462. Just a weird corner case.

Reply:
Oh, I had not connected those dots. I suppose that begs the question "What should the default be?". Hopefully sticking with what was defined in the `ini` file will be ok. I think that was effectively the default anyway since the script before would not run without an `ini` file.

Thanks for the feedback.